### PR TITLE
Run tests

### PR DESCRIPTION
**Pull request by moe-irfan**
- Runs jest tests on every commit on all branches.
- Runs jest tests on merged results if we're inside of a pull request.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Test
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
```
Comment (irfan798) [inline .github/workflows/test.yml]: Can we use something more lightweight and deterministic?
latest can change any moment

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Run test over the merged changes
+        run: |
+            git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+            git config --global user.name "github-actions[bot]"
+            git fetch origin $GITHUB_BASE_REF
+            git checkout -b pr
+            git checkout $GITHUB_BASE_REF
+            git merge --no-commit --no-edit --allow-unrelated-histories pr
+            yarn test
```
Comment (irfan798) [inline .github/workflows/test.yml:32]: add 
- `yarn`
- `yarn test`

Because package.json could have changed

**Review by irfan798 — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (moe-irfan) [reply .github/workflows/test.yml:32]: Good catch! Added it.

File: .github/workflows/test.yml (hunk 1 above)
Comment (moe-irfan) [reply .github/workflows/test.yml]: I don't see alpine under https://github.com/actions/runner-images 😄 

Pinned it to latest LTS though. 22.04

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Run test over the merged changes
+        run: |
+            git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+            git config --global user.name "github-actions[bot]"
+            git fetch origin $GITHUB_BASE_REF
+            git checkout -b pr
+            git checkout $GITHUB_BASE_REF
+            git merge --no-commit --no-edit --allow-unrelated-histories pr
+            yarn
+            yarn test
```
Comment (irfan798) [inline .github/workflows/test.yml:32]: we also forgot to add` yarn lerna run build` after `yarn`  😀

Otherwise tests will always fail

**Review by irfan798 — CHANGES_REQUESTED** [review]

**Review by irfan798 — APPROVED** [review]

**Pull request merged**